### PR TITLE
Prototype test script for TTINTEL-239

### PR DESCRIPTION
**Pull request by spenced**

### CODE REVIEW

**Review by jnunyez — COMMENTED** [review]
Went through the code, got one big question: 
What is the value of the FSM for the post-processor? (class FSM). Is it maybe aiming at identify root cause of a problem? I really would like to understand the benefit (I'm sure there is).

File: prototype.py [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    fsm.state = STATE_TESTING
+    fsm.state = STATE_TESTING_UNKNOWN
+
+def handle_S0_in_TESTING(fsm, arg):
+    """Handle event S0 in state /TESTING"""
+    fsm.state = STATE_TESTING_INITIALIZING
+
+def handle_S0_in_TESTING_INITIALIZING(fsm, arg):
+    """Handle event S0 in state /TESTING/INITIALIZING"""
+    fsm.state = STATE_TESTING_INITIALIZING
```
[the remark this answers is not in the document]
Comment (spenced) [reply prototype.py:39]: The state assignments are output on each "state boundary crossing" for an external transition. An external transition is one where the FSM leaves the current state before entering the target state. The state boundaries crossed are all those up to (but not including) the common ancestor of current and target states. With regard to "assigning `fsm.state` to the value it already has", that will be due to an external transition back into the same state.

Currently I expect these to be optimized away (by Python itself). I've left them in the generated output as they document the boundary crossings, allowing visual inspection that the intended path is taken.

I agree they're ugly. Getting rid of them is on my TODO list.

**Comment by spenced** [comment]
Replying to Jose's comment... for this prototype, the purpose was twofold: 1) to make it easy for me to identify events so I could draw vertical lines in the plot; 2) to stimulate conversation around separation of concerns, how we combine code to produce higher order tests.

**Comment by jnunyez** [comment]
> Replying to Jose's comment... for this prototype, the purpose was twofold: 1) to make it easy for me to identify events so I could draw vertical lines in the plot; 2) to stimulate conversation around separation of concerns, how we combine code to produce higher order tests.

@crwr45 @spenced Propose to  close this PR since this work will be moved to https://github.com/redhat-partner-solutions/vse-sync-pp

**Pull request closed without merging**